gateway-framework/src/network/indexing_performance.rs

```diff
@@ -1,14 +1,19 @@
 use std::{collections::HashMap, ops::Deref, time::Duration};
 
-use alloy_primitives::BlockNumber;
+use alloy_primitives::{Address, BlockNumber};
 use eventuals::{Closed, Eventual, Ptr};
+use thegraph_core::types::DeploymentId;
 use tokio::{
-    select,
+    self,
     sync::{mpsc, oneshot, RwLock},
-    time::{self, MissedTickBehavior},
+    time::MissedTickBehavior,
 };
 
-use crate::{indexing::Indexing, network::discovery::Status};
+#[derive(Clone)]
+pub struct Status {
+    /// The latest indexed block
```

Comment:
I prefer the term `reported` block rather than `indexed` block to emphasize that we don't trust it at face value.

Reply:
Addressed in [30a6b1b](https://github.com/edgeandnode/gateway/commit/30a6b1b7a049ba7f7c4721b386ea110613e64ced)